src/anndata_to_avro.py

```diff
@@ -0,0 +1,205 @@
+                u'cas_feature_index': cas_feature_index.item(),
+                u'raw_counts': v_int
+            }
+
+    write_avro(raw_counts_generator, parsed_schema, filename)
+
+
+def dump_cell_info(adata, filename, cas_cell_index_start):
+    # Dump out cell info (obs) -- cell_id is index (26 columns).
+    adata.obs['original_cell_id'] = adata.obs.index
+    adata.obs['cas_cell_index'] = np.arange(cas_cell_index_start, cas_cell_index_start + len(adata.obs))
+
+    row_index_to_cas_cell_index = [None] * len(adata.obs)
+    for row in range(0, len(adata.obs)):
+        row_index_to_cas_cell_index[row] = adata.obs['cas_cell_index'].iloc[[row]][0]
+
+    schema = {
+        'doc': 'CAS Cell Info',
+        'namespace': 'cas',
+        'name': 'CellInfo',
+        'type': 'record',
+        'fields': [
+            {'name': 'cas_cell_index', 'type': 'int'},
+            {'name': 'original_cell_id', 'type': 'string'},
+            {'name': 'cell_type', 'type': 'string'}
+        ]
+    }
+
+    parsed_schema = parse_schema(schema)
+
+    def cell_generator():
+        cells = adata.obs[['cas_cell_index', 'original_cell_id', 'cell_type']]
+        for _, r in cells.iterrows():
+            yield {
+                u'cas_cell_index': r['cas_cell_index'],
+                u'original_cell_id': r['original_cell_id'],
+                u'cell_type': r['cell_type']
+            }
+
+    write_avro(cell_generator, parsed_schema, filename)
+
+    return row_index_to_cas_cell_index
+
+
+def dump_feature_info(adata, filename, cas_feature_index_start):
+    # Dump out feature info -- feature_id is index (12 columns).
+    adata.var['original_feature_id'] = adata.var.index
+
+    adata.var['cas_feature_index'] = np.arange(cas_feature_index_start, cas_feature_index_start + len(adata.var))
+    col_index_to_cas_feature_index = [None] * len(adata.var)
+    for col in range(0, len(adata.var)):
+        col_index_to_cas_feature_index[col] = adata.var['cas_feature_index'].iloc[[col]][0]
+
+    schema = {
+        'doc': 'CAS Feature Info',
+        'namespace': 'cas',
+        'name': 'FeatureInfo',
+        'type': 'record',
+        'fields': [
+            {'name': 'cas_feature_index', 'type': 'int'},
+            {'name': 'original_feature_id', 'type': 'string'},
+            {'name': 'feature_name', 'type': 'string'}
+        ]
+    }
+
+    parsed_schema = parse_schema(schema)
+
+    def feature_generator():
+        features = adata.var[['cas_feature_index', 'original_feature_id', 'feature_name']]
+        for _, row in features.iterrows():
+            yield {
+                u'cas_feature_index': row['cas_feature_index'],
+                u'original_feature_id': row['original_feature_id'],
+                u'feature_name': row['feature_name']
+            }
+
+    write_avro(feature_generator, parsed_schema, filename)
+    return col_index_to_cas_feature_index
+
+
+def check_output_files(avro_prefix):
```

Comment:
nit: clearer name as to what this is doing (check output files don't exist) or something?